### PR TITLE
feat: add compaction level to commit metrics

### PR DESCRIPTION
**Pull request by crepererum**

### CODE REVIEW

File: compactor2/src/components/commit/metrics.rs [hunk 1]
```diff
@@ -284,19 +426,68 @@ mod tests {
...
+                METRIC_NAME_FILE_BYTES,
+                [("op", "create"), ("level", "L0")]
+            ),
+            1
+        );
+        assert_eq!(
+            hist_count(
+                &registry,
+                METRIC_NAME_FILE_BYTES,
+                [("op", "upgrade"), ("from", "L1"), ("to", "L1")]
```
Comment (NGA-TRAN) [inline compactor2/src/components/commit/metrics.rs:441]: When we upgrad, the level will move up. We want the test to reflect that even if we can make them anything. Either one from L0 to L1 or from L1 to L2

File: compactor2/src/components/commit/metrics.rs [hunk 2]
```diff
@@ -284,19 +426,68 @@ mod tests {
...
+            ),
             10_016
         );
         assert_eq!(
-            hist_total(&registry, METRIC_NAME_FILE_BYTES, "upgrade"),
-            20_010
+            hist_total(
+                &registry,
+                METRIC_NAME_FILE_BYTES,
+                [("op", "upgrade"), ("from", "L1"), ("to", "L1")]
```
Comment (NGA-TRAN) [inline compactor2/src/components/commit/metrics.rs:473]: same here

**Review by NGA-TRAN — APPROVED** [review]
Nice. only 2 tests need to update to reflect meaningful use cases

File: compactor2/src/components/commit/metrics.rs (hunk 1 above)
Comment (crepererum) [reply compactor2/src/components/commit/metrics.rs]: done

File: compactor2/src/components/commit/metrics.rs (hunk 2 above)
Comment (crepererum) [reply compactor2/src/components/commit/metrics.rs]: done

**Pull request merged**